respeecher_tts/api.py

```diff
@@ -45,21 +45,26 @@ def _get_voices(self) -> list[Voice]:
 
     @cache
     def _lookup_voice_and_ns(
-        self, voice_name: str, narration_style_name: str | None = None
+        self, voice_name: str, narration_style_tags: tuple[str] | None = None
     ) -> tuple[str, str]:
         voice = next((v for v in self.voices if v.name == voice_name), None)
         if not voice:
             raise ValueError(f"Voice {voice_name} not found")
 
-        if narration_style_name:
-            narration_style = next(
-                (
-                    ns
-                    for ns in voice.narration_styles
-                    if ns.info.name == narration_style_name
-                ),
-                None,
-            )
+        if narration_style_tags:
+            # search for the narration style that has the most tags in common
+            # with narration_style_tags
+            best_match = 0
+            narration_style = None
```

Comment:
If we have not found matching tags -> we will have narration_style = None -> we will raise ValueError on line 73. We need to change message for it